bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/handler/ModuleHandler.java

```diff
@@ -25,6 +25,16 @@
 @NonNullByDefault
 public interface ModuleHandler {
 
+    /**
+     * Initializes the module handler, e.g. schedules jobs.
+     * <p>
+     * This method is only called, if the {@link ModuleHandlerCallback} is set.
+     * <p>
+     * The framework expects this method to be non-blocking and return quickly. For longer running initializations,
+     * the implementation has to take care of scheduling a separate job.
+     */
+    public void initialize();
```

Comment:
Why do you introduce this method? Can't the handler initialize stuff when the callback is set?
I don't really see the need for both methods. We could also change the existing `setCallback` to be `initialize` with the callback as a parameter, but this would be a breaking change.

Reply:
You're right that's redundant. I removed the initialize method again.